lib/galaxy/datatypes/flow.py

```diff
@@ -0,0 +1,435 @@
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Text Flow file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on file formatting and values"""
+        with open(filename, "r") as f:
+            f.readline()
+            values = f.readline().strip().split("\t")
+            for vals in values:
+                if not is_number(vals):
+                    return False
+            return True
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'text/tab-separated-values'
+
+
+class FlowClustered(Tabular):
+    """Class describing a Flow Text that has been clustered through FLOCK"""
+    file_ext = "flowclr"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Text Flow Clustered file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Flow Text Clustered file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on headers and values"""
+        with open(filename, "r") as f:
+            population = f.readline().strip().split("\t")[-1]
+            if population != "Population":
+                return False
+            values = f.readline().strip().split("\t")
+            for vals in values:
+                if not is_number(vals):
+                    return False
+            return True
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'text/tab-separated-values'
+
+
+class FlowMFI(Tabular):
+    """Class describing a Flow MFI file"""
+    file_ext = "flowmfi"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "MFI Flow file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "MFI Flow file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on file formatting and values"""
+        with open(filename, "r") as f:
+            population = f.readline().strip().split("\t")[0]
+            if population != "Population":
+                return False
+            values = f.readline().strip().split("\t")
+            for vals in values:
+                if not is_number(vals):
+                    return False
+            return True
```

Comment:
Same as flowclustered?

Reply:
This one also request in the previous lines to have `Population_ID` as first token of the header.

Reply:
Please let me know if I should remove this as well then @mvdbeek. Thanks